llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp

```diff
@@ -240,6 +254,10 @@ RISCVRegisterBankInfo::getInstrMapping(const MachineInstr &MI) const {
       &RISCV::ValueMappings[GPRSize == 64 ? RISCV::GPRB64Idx
                                           : RISCV::GPRB32Idx];
 
+  unsigned VRBSize = getMaximumSize(RISCV::VRBRegBankID);
+  const ValueMapping *VRBValueMapping =
+      &RISCV::ValueMappings[VRBSize == 64 ? RISCV::VRB64Idx : RISCV::VRB32Idx];
```

Comment:
The size for VRB bank shouldn't change. We only have to do this for GPR because of RV32 vs RV64.

Reply:
for VRB, do we always have size == 64?

Reply:
It looks like in `build/lib/Target/RISCV/RISCVGenRegisterBank.inc` that the `Sizes` array has 512 for both RV32 and RV64 for VRB.

Reply:
512 is the size of LMUL=8. we should have 4 different sizes in the mapping for the 4 different whole LMUL register sizes. We should pick based on the known minimum size of the type. 512, 256, 128, and 64. Anything less than 64 should use the 64.

Reply:
Actually I'm a bit lost right now. Can I check my understanding?

1. 
> The size for VRB bank shouldn't change.

Does this refer to `unsigned VRBSize` never change?

2. Based on my understanding of your description, in [`PartMappings`](https://github.com/llvm/llvm-project/blob/c556c9c3792a322dfc3a9c60be9d7d32cf335235/llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp#L27), I'll probably have:
```C++
    {0, 64, VRBRegBank}, // <= 64
    {0, 64, VRBRegBank}, {64, 128, VRBRegBank}, // 128, LMUL=2
    {0, 64, VRBRegBank}, {64, 128, VRBRegBank}, {128, 192, VRBRegBank}, {192, 256, VRBRegBank}, // 256, LMUL=4
    {0, 64, VRBRegBank} .... // 512, LMUL=8
```
?
And that I'll only have `PMI_VRB64` in [`PartialMappingIdx`](https://github.com/llvm/llvm-project/blob/c556c9c3792a322dfc3a9c60be9d7d32cf335235/llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp#L34) since the size of VRB doesn't change?
And in [`ValueMapping`](https://github.com/llvm/llvm-project/blob/c556c9c3792a322dfc3a9c60be9d7d32cf335235/llvm/lib/Target/RISCV/GISel/RISCVRegisterBankInfo.cpp#L41), I'll probably have:
```C++
{&PartMappings[PMI_VRB64], 1},
{&PartMappings[PMI_VB64], 2},
{&PartMappings[PMI_VRB64], 4},
{&PartMappings[PMI_VRB64], 8}
```
?